### PR TITLE
Fix textual updates, new integration test suite for flutter

### LINKED ISSUE #1053: Text updates only creates black/emptty update
Image

### PR DESCRIPTION
**Pull request by gnunicorn**
Fixes #1053 .

Also includes a revamp of the integration test suite from gherkin (fully removed) and replaced with `convenient_test`. Next to login and registration smoketest, there is an additional tests creating a space and posting a textual news item - proofing that it works now (and will continue to do so). In order to make that happen, more `Keys` are introduced throughout the UI, and various smaller fixes are made to allow for items to be clickable with the UI runners. Additionally, the PR contains documentation on how to use the test runner yourself.

Open Tasks:
 - [ ] github workflows run on requests and on `main` (already in but defunct). 
 - [x] can we create api docs for the tester?
 - [x] the rescaling issue in the onboarding screen is fixed in a wrong way that now makes the font super small even if the space is given.

### CODE REVIEW

File: app/lib/common/providers/common_providers.dart [hunk 1]
```diff
@@ -32,6 +32,15 @@ Future<ProfileData> getProfileData(Account account) async {
   return ProfileData(displayName.text(), avatar.data());
 }
 
+final myUserIdStrProvider = StateProvider((ref) {
+  final client = ref.watch(clientProvider);
```
Comment (gtalha07) [inline app/lib/common/providers/common_providers.dart:36]: Why aren't we using ```alwaysClientProvider``` here ??

File: app/lib/features/events/sheets/create_event_sheet.dart [hunk 2]
```diff
@@ -209,9 +209,9 @@ class _CreateEventSheetConsumerState extends ConsumerState<CreateEventSheet> {
       draft.utcEndFromRfc3339(utcEndDateTime);
       draft.descriptionText(description);
       final eventId = await draft.send();
-      final client = ref.read(clientProvider);
+      final client = ref.read(alwaysClientProvider);
```
Comment (gtalha07) [inline app/lib/features/events/sheets/create_event_sheet.dart:212]: Good to have improved it, but why not using it everywhere else?

Comment (gnunicorn) [reply app/lib/features/events/sheets/create_event_sheet.dart:212]: I only tried to improve the problem that the integration tests failed with `null!` issues and replaced it where I found that pattern. We can extend that to other places, but it is only a softer version of it, so places where this throwing isn't acceptable, isn't good...

File: app/lib/common/providers/common_providers.dart (hunk 1 above)
Comment (gnunicorn) [reply app/lib/common/providers/common_providers.dart:36]: came up with it after, but we could ;)

File: app/lib/features/onboarding/pages/login_page.dart [hunk 3]
```diff
@@ -54,7 +54,9 @@ class _LoginPageState extends ConsumerState<LoginPage> {
           // no message means, login was successful.
           context.goNamed(Routes.main.name);
         } else {
-          customMsgSnackbar(context, loginSuccess);
+          EasyLoading.showError(
```
Comment (gtalha07) [inline app/lib/features/onboarding/pages/login_page.dart:57]: could use ```showSuccess``` here.

**Review by gtalha07 — APPROVED** [review]
Looks sleek and good to merge!! Except few commented parts which can be addressed on your behalf ;)

File: app/lib/features/onboarding/pages/login_page.dart (hunk 3 above)
Comment (gnunicorn) [reply app/lib/features/onboarding/pages/login_page.dart:57]: no... if we have a message it is an "error message" ... this is a bit confusing in the API but that's how it is ...

**Pull request merged**